### PR TITLE
fix: selectedOption not working properly. unable to…

### PR DESCRIPTION
**Pull request by biplavmz**
…  empty the selectedoption useState while giving input to the search onscroll unnecessary data loading because of that wrong data mapin. i have fix this search issues 
#1585

## Description

In search input while searching for book keyword  we get 3 book related results  while on scroll this result get changes we are getting unnecessary result because of that our result get changes and unnecessary api call.
in code if you see we are unable to empty the  selectedOptions because of that some tag or search keyword triggering. I have fix this issue please check and merge my pr.
## Screenshots

Screenshot 2023-07-05 131815

### CODE REVIEW

File: package-lock.json
Comment (priyankarpal) [inline package-lock.json:1]: remove the line & add `package-lock.json` to the `.gitignore` file

**Review by priyankarpal — CHANGES_REQUESTED** [review]

File: app/projects/page.tsx [hunk 1]
```diff
@@ -144,17 +153,16 @@ const ProjectsPage: NextPage = () => {
       setVisibleProjects(Projects.slice(0, 15));
       return;
     }
-
-    const currProjects = getFilteredProjects();
-    setVisibleProjects(currProjects);
+    // const currProjects = getFilteredProjects();
+    // setVisibleProjects(currProjects);
```
Comment (priyankarpal) [inline app/projects/page.tsx:157]: remove the unwanted comments

**Review by priyankarpal — CHANGES_REQUESTED** [review]

**Comment by priyankarpal** [comment]
@biplavmz avoid creating PR first

**Review by biplavmz — COMMENTED** [review]
i have fix the issue please check and merge

**Comment by biplavmz** [comment]
Thanks @priyankarpal for merged my Pr

**Pull request merged**